pkg/reconciler/trustedresources/verify_test.go

```diff
@@ -94,6 +102,186 @@ func TestVerifyInterface_Task(t *testing.T) {
+	if err != nil {
+		t.Fatal("fail to sign task", err)
+	}
+
+	tamperedPipeline := signedPipeline.DeepCopy()
+	tamperedPipeline.Annotations["random"] = "attack"
+
+	tcs := []struct {
+		name     string
+		pipeline v1beta1.PipelineObject
+		wantErr  bool
+	}{{
+		name:     "Signed Pipeline Passes Verification",
+		pipeline: signedPipeline,
+		wantErr:  false,
+	}, {
+		name:     "Tampered Pipeline Fails Verification with tampered content",
+		pipeline: tamperedPipeline,
+		wantErr:  true,
+	}, {
+		name:     "Unsigned Pipeline Fails Verification without signature",
+		pipeline: unsignedPipeline,
+		wantErr:  true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			err := VerifyPipeline(ctx, tc.pipeline)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("VerifyPipeline() get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+func TestPrepareObjectMeta(t *testing.T) {
+	unsigned := ttesting.GetUnsignedTask("test-task").ObjectMeta
+
+	signed := unsigned.DeepCopy()
+	signed.Annotations = map[string]string{SignatureAnnotation: "tY805zV53PtwDarK3VD6dQPx5MbIgctNcg/oSle+MG0="}
+
+	signedWithLabels := signed.DeepCopy()
+	signedWithLabels.Labels = map[string]string{"label": "foo"}
+
+	signedWithExtraAnnotations := signed.DeepCopy()
+	signedWithExtraAnnotations.Annotations["kubectl-client-side-apply"] = "client"
+	signedWithExtraAnnotations.Annotations["kubectl.kubernetes.io/last-applied-configuration"] = "config"
+
+	tcs := []struct {
+		name       string
+		objectmeta *metav1.ObjectMeta
+		expected   metav1.ObjectMeta
+		wantErr    bool
+	}{{
+		name:       "Prepare signed objectmeta without labels",
+		objectmeta: signed,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Annotations: map[string]string{},
+		},
+		wantErr: false,
+	}, {
+		name:       "Prepare signed objectmeta with labels",
+		objectmeta: signedWithLabels,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Labels:      map[string]string{"label": "foo"},
+			Annotations: map[string]string{},
+		},
+		wantErr: false,
+	}, {
+		name:       "Prepare signed objectmeta with extra annotations",
+		objectmeta: signedWithExtraAnnotations,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Annotations: map[string]string{},
+		},
+		wantErr: false,
+	}, {
+		name:       "Fail preparation without signature",
+		objectmeta: &unsigned,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Annotations: map[string]string{"foo": "bar"},
+		},
+		wantErr: true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			task, signature, err := prepareObjectMeta(*tc.objectmeta)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("prepareObjectMeta() get err %v, wantErr %t", err, tc.wantErr)
+			}
+			if d := cmp.Diff(task, tc.expected); &tc.expected != nil && d != "" {
```

Comment:
I don't think `&tc.expected` will ever be nil since it started as a non-pointer object, so you can remove it and just check if `d != ""` (I think `cmp` is also smart enough to check for nil vs non-nil values)

Reply:
Thanks!